gpiozero/output_devices.py

```diff
@@ -1231,6 +1261,22 @@ def value(self, value):
         else:
             self.stop()
 
+    @property
+    def has_enable(self):
```

Comment:
I wonder if it might make sense to hoist these properties up into the `ControlType` class?
I've not used the `Enum` type in Python before, but after a quick bit of experimentation, the following _seems_ to work?

```python3
class ControlType(Enum):
    DIGITAL = auto()
    PWM = auto()
    DIGITAL_WITH_DIGITAL_ENABLE = auto()
    PWM_WITH_DIGITAL_ENABLE = auto()
    DIGITAL_WITH_PWM_ENABLE = auto()

    @property
    def has_pwm(self):
        return 'PWM' in self.name

    @property
    def has_enable(self):
        return 'ENABLE' in self.name
```